modules/fx/arrow-fx/src/main/kotlin/arrow/fx/Queue.kt

```diff
@@ -198,6 +208,61 @@ class Queue<F, A> private constructor(private val capacity: Int, private val ref
         ifShutdown = ::identity
       )
     }
+
+  private sealed class State<F, out A> {
+    abstract fun size(): Kind<F, Int>
+
+    internal data class Deficit<F, A>(val takers: IQueue<Promise<F, A>>, val AP: Applicative<F>, val shutdownHook: Kind<F, Unit>) : State<F, A>() {
```

Comment:
Same here. I missed this last time.
```suggestion
    data class Deficit<F, A>(val takers: IQueue<Promise<F, A>>, val AP: Applicative<F>, val shutdownHook: Kind<F, Unit>) : State<F, A>() {
```